File: task3/.cs
using System;                      
using System.Collections.Generic; 
using System.Linq;


public class Program
{
    static bool IsAscending(int n)   // Проверяет, возрастают ли цифры числа
    {
        string s = n.ToString();  // Число в строку
        for (int i = 1; i < s.Length; i++) // Цикл по цифрам
        {
            if (s[i] <= s[i - 1])     // Если не возрастает
            {
                return false;       // Не возрастающее
            }
        }
        return true;               // Возрастающее
    }

    static int ReverseNumber(int n)  // Переворачивает число
    {
        string s = n.ToString();  // Число в строку
        return int.Parse(new string(s.Reverse().ToArray())); // Переворачиваем строку и обратно в число
    }

    public static void Main(string[] args) // Главная функция
    {
        List<int> numbers = new List<int> { 4, 87, 129, 33, 45 }; // Список чисел
        Console.Write("Result1: "); // Вывод текста
        foreach (int num in numbers)    // Цикл по числам
        {
            if (IsAscending(num))      // Если возрастающее
            {
                Console.Write(ReverseNumber(num) + " "); // Вывод перевернутого числа
            }
        }
        Console.WriteLine();          // Перенос строки
    }
}

File: task3/.rb
def is_ascending(n) # Проверяет, возрастают ли цифры числа
  s = n.to_s # Преобразует число в строку
  (1...s.length).each do |i| # Итерируется по цифрам (начиная со второй)
    return false if s[i].to_i <= s[i-1].to_i # Если текущая цифра меньше или равна предыдущей, возвращает false
  end
  true # Если все цифры возрастают, возвращает true
end

def reverse_number(n) # Переворачивает число
  n.to_s.reverse.to_i # Преобразует в строку, переворачивает, преобразует обратно в число
end

numbers = [4, 87, 129, 33, 45] # Массив чисел
print "Результат: " # Выводит "Результат: "
numbers.each do |num| # Итерируется по массиву чисел
  if is_ascending(num) # Если число возрастающее
    print "#{reverse_number(num)} " # Выводит перевернутое число с пробелом
  end
end
puts # Выводит перенос строки
